presto-hive/src/test/java/com/facebook/presto/hive/metastore/thrift/TestThriftMetastoreUtil.java

```diff
@@ -341,6 +341,28 @@ public void testEmptyBinaryStatsToColumnStatistics()
         assertEquals(actual.getDistinctValuesCount(), OptionalLong.empty());
     }
 
+    @Test
+    public void testSingleDistinctValue()
+    {
+        DoubleColumnStatsData doubleColumnStatsData = new DoubleColumnStatsData();
+        doubleColumnStatsData.setNumNulls(10);
+        doubleColumnStatsData.setNumDVs(1);
+        ColumnStatisticsObj columnStatisticsObj = new ColumnStatisticsObj("my_col", DOUBLE_TYPE_NAME, doubleStats(doubleColumnStatsData));
+        HiveColumnStatistics actual = fromMetastoreApiColumnStatistics(columnStatisticsObj, OptionalLong.of(10));
+
+        assertEquals(actual.getNullsCount(), OptionalLong.of(10));
+        assertEquals(actual.getDistinctValuesCount(), OptionalLong.of(0));
+
+        doubleColumnStatsData = new DoubleColumnStatsData();
+        doubleColumnStatsData.setNumNulls(10);
+        doubleColumnStatsData.setNumDVs(1);
+        columnStatisticsObj = new ColumnStatisticsObj("my_col", DOUBLE_TYPE_NAME, doubleStats(doubleColumnStatsData));
+        actual = fromMetastoreApiColumnStatistics(columnStatisticsObj, OptionalLong.of(11));
+
+        assertEquals(actual.getNullsCount(), OptionalLong.of(10));
+        assertEquals(actual.getDistinctValuesCount(), OptionalLong.of(1));
```

Comment:
If number of nulls is equal to number of rows, the `NDV` value must be `0`. Since there is not a single non null value.